### ISSUE TITLE
[Fine Tune] Self-made training dataset results in Loss = 0

### ISSUE THREAD

**Issue by piqiuni** [QwenLM/Qwen3-VL#1981]
Hello,
I am fine-tuning Qwen3VL-4B-Instruct using a self-made dataset.
During the training process, I encountered Loss=0.
After investigation, I found that adding or removing one row from the dataset causes it to be untrainable or trainable.
Please help me analyze this.
> {'loss': 2.7584, 'grad_norm': 1.4142135381698608, 'learning_rate': 0.0, 'epoch': 0.02}                                                                                        
> {'loss': 0.0, 'grad_norm': 1.4142135381698608, 'learning_rate': 5e-06, 'epoch': 0.05}                                                                                         
> {'loss': 0.0, 'grad_norm': 1.4142135381698608, 'learning_rate': 1e-05, 'epoch': 0.07}
Dataset causing Loss=0
```
{
    "image": "sat_z15_30.884667_121.192373/sat_30.884667_121.192373_z15_s2.png",
    "conversations": [
      {
        "from": "human",
        "value": "<image>\n任务:步行路径规划\n起点像素: (774, 526), 终点像素: (385, 699)\n输出要求:1.根据图像规划1-3条不同路径2.每条路径必须包含：距离(米)、预计时间(秒)、详细的文字导航指令、以及路径像素坐标点数组\n"
      },
      {
        "from": "gpt",
        "value": "路径1:\n距离: 1504 米，预计时间: 1203 秒\n\n路径2:\n距离: 1574 米，预计时间: 1259 秒\n\n路径3:\n距离: 1623 米，预计时间: 1298 秒"
      }
    ]
  },
```

Trainable data samples
```
{
    "image": "sat_z15_30.884667_121.192373/sat_30.884667_121.192373_z15_s2.png",
    "conversations": [
      {
        "from": "human",
        "value": "<image>\n任务:步行路径规划\n起点像素: (774, 526), 终点像素: (385, 699)\n输出要求:1.根据图像规划1-3条不同路径\n"
      },
      {
        "from": "gpt",
        "value": "路径1:\n距离: 1504 米，预计时间: 1203 秒\n\n路径2:\n距离: 1574 米，预计时间: 1259 秒\n\n路径3:\n距离: 1623 米，预计时间: 1298 秒"
      }
    ]
  },
```

**Comment by JJJYmmm** [comment]
maybe related to zero2/zero3 https://github.com/QwenLM/Qwen3-VL/issues/1658#issuecomment-3601158530

**Comment by piqiuni** [comment]
> maybe related to zero2/zero3 [#1658 (comment)](https://github.com/QwenLM/Qwen3-VL/issues/1658#issuecomment-3601158530)

Change "overlap_comm": true to false solved the problem
Thanks a lot.

I also met a problem that the loss value of zero2 and zero3 training are different, do you have any idea?
https://www.reddit.com/r/deeplearning/comments/1qju4rk/deepspeed_zero2_and_zero3_training_got_different/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button
